finetuner/client/client.py

```diff
@@ -175,4 +194,47 @@ def get_run_status(self, experiment_name: str, run_name: str):
             / run_name
             / STATUS
         )
-        return self.handle_request(url=url, method=GET, params={USER_ID: self._user_id})
+        return self.handle_request(url=url, method=GET)
+
+    def _push_data_to_hubble(
+        self,
+        train_data: Union[DocumentArray, str],
+        eval_data: Optional[Union[DocumentArray, str]],
+        experiment_name: str,
+        run_name: str,
+    ):
+        """Push DocumentArray for training and evaluation data on Hubble if it's not already uploaded.
+
+        :param train_data: Either a `DocumentArray` for training data that needs to be pushed on Hubble
+                          or a name of the `DocumentArray` that is already uploaded.
+        :param eval_data: Either a `DocumentArray` for evaluation data that needs to be pushed on Hubble
+                          or a name of the `DocumentArray` that is already uploaded.
+        :returns: Name(s) of pushed `DocumentArray`-s.
+        """
+        if isinstance(train_data, DocumentArray):
+            da_name = '-'.join(
+                [self._hubble_user_id, experiment_name, run_name, TRAIN_DATA]
+            )
+            train_data.push(name=da_name)
+            train_data = da_name
+        if isinstance(eval_data, DocumentArray):
+            da_name = '-'.join(
+                [self._hubble_user_id, experiment_name, run_name, EVAL_DATA]
+            )
+            eval_data.push(name=da_name)
+            eval_data = da_name
```

Comment:
let's first assume they're always `DocumentArray`, not `str`, otherwise `str` is not handled here

Reply:
and `eval_data` is `Optional`, make sure it is checked `if eval_data and isinstance...`

Reply:
Sure but it works as expected - passing [these tests](https://github.com/jina-ai/finetuner/blob/2b8625ac62c0db02b86f49148e51ddbfb17b8ba3/tests/unit/test_hubble.py#L7) (One thing that's missing for strings is checking if the name is valid) - so maybe better to leave it as it is for now and refactor it later?

Added `if eval_data ...`